executor/split_test.go

```diff
@@ -0,0 +1,298 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"bytes"
+	"encoding/binary"
+	"math"
+	"math/rand"
+
+	. "github.com/pingcap/check"
+	"github.com/pingcap/parser/model"
+	"github.com/pingcap/parser/mysql"
+	"github.com/pingcap/tidb/table/tables"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/mock"
+)
+
+var _ = Suite(&testSplitIndex{})
+
+type testSplitIndex struct {
+}
+
+func (s *testSplitIndex) SetUpSuite(c *C) {
+}
+
+func (s *testSplitIndex) TearDownSuite(c *C) {
+}
+
+func (s *testSplitIndex) TestLongestCommonPrefixLen(c *C) {
+	cases := []struct {
+		s1 string
+		s2 string
+		l  int
+	}{
+		{"", "", 0},
+		{"", "a", 0},
+		{"a", "", 0},
+		{"a", "a", 1},
+		{"ab", "a", 1},
+		{"a", "ab", 1},
+		{"b", "ab", 0},
+		{"ba", "ab", 0},
+	}
+
+	for _, ca := range cases {
+		re := longestCommonPrefixLen([]byte(ca.s1), []byte(ca.s2))
+		c.Assert(re, Equals, ca.l)
+	}
+}
+
+func (s *testSplitIndex) TestgetStepValue(c *C) {
+	cases := []struct {
+		lower []byte
+		upper []byte
+		l     int
+		v     uint64
+	}{
+		{[]byte{}, []byte{}, 0, math.MaxUint64},
+		{[]byte{0}, []byte{128}, 0, binary.BigEndian.Uint64([]byte{128, 255, 255, 255, 255, 255, 255, 255})},
+		{[]byte{'a'}, []byte{'z'}, 0, binary.BigEndian.Uint64([]byte{'z' - 'a', 255, 255, 255, 255, 255, 255, 255})},
+		{[]byte("abc"), []byte{'z'}, 0, binary.BigEndian.Uint64([]byte{'z' - 'a', 255 - 'b', 255 - 'c', 255, 255, 255, 255, 255})},
+		{[]byte("abc"), []byte("xyz"), 0, binary.BigEndian.Uint64([]byte{'x' - 'a', 'y' - 'b', 'z' - 'c', 255, 255, 255, 255, 255})},
+		{[]byte("abc"), []byte("axyz"), 1, binary.BigEndian.Uint64([]byte{'x' - 'b', 'y' - 'c', 'z', 255, 255, 255, 255, 255})},
+		{[]byte("abc0123456"), []byte("xyz01234"), 0, binary.BigEndian.Uint64([]byte{'x' - 'a', 'y' - 'b', 'z' - 'c', 0, 0, 0, 0, 0})},
+	}
+
+	for _, ca := range cases {
+		l := longestCommonPrefixLen(ca.lower, ca.upper)
+		c.Assert(l, Equals, ca.l)
+		v0 := getStepValue(ca.lower[l:], ca.upper[l:], 1)
+		c.Assert(v0, Equals, ca.v)
+	}
+}
+
+func (s *testSplitIndex) TestSplitIndex(c *C) {
+	tbInfo := &model.TableInfo{
+		Name: model.NewCIStr("t1"),
+		ID:   rand.Int63(),
+		Columns: []*model.ColumnInfo{
+			{
+				Name:         model.NewCIStr("c0"),
+				ID:           1,
+				Offset:       1,
+				DefaultValue: 0,
+				State:        model.StatePublic,
+				FieldType:    *types.NewFieldType(mysql.TypeLong),
+			},
+		},
+	}
+	idxCols := []*model.IndexColumn{{Name: tbInfo.Columns[0].Name, Offset: 0, Length: types.UnspecifiedLength}}
+	idxInfo := &model.IndexInfo{
+		ID:      1,
+		Name:    model.NewCIStr("idx1"),
+		Table:   model.NewCIStr("t1"),
+		Columns: idxCols,
+		State:   model.StatePublic,
+	}
+
+	// Test for int index.
+	// range is 0 ~ 100, and split into 10 region.
+	// So 10 regions range is like below:
+	// region1: -inf ~ 10
+	// region2: 10 ~ 20
```

Comment:
Could you add symbols about open and closed intervals?

Reply:
Good catch. 
done.